libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2252,3 +2252,102 @@ string YulUtilFunctions::revertReasonIfDebug(string const& _message)
 {
 	return revertReasonIfDebug(m_revertStrings, _message);
 }
+
+string YulUtilFunctions::tryDecodeErrorMessageFunction()
+{
+	string const functionName = "try_decode_error_message";
+
+	return m_functionCollector.createFunction(functionName, [&]() {
+		string const errorHash = FixedHash<4>(util::keccak256("Error(string)")).hex();
+		return util::Whiskers(R"(
+			function <functionName>() -> data {
+				data := mload(<freeMemoryPointer>)
+
+				if lt(returndatasize(), 0x44) {
+					data := 0
+					leave
+				}
+
+				returndatacopy(0, 0, 4)
+				let sig := <getSig>
+				if iszero(eq(sig, 0x<ErrorSignature>)) {
+					data := 0
+					leave
+				}
+
+				returndatacopy(data, 4, sub(returndatasize(), 4))
+				let offset := mload(data)
+				if or(
+					gt(offset, 0xffffffffffffffff),
+					gt(add(offset, 0x24), returndatasize())
+				) {
+					data := 0
+					leave
+				}
+
+				let msg := add(data, offset)
+				let length := mload(msg)
+				if gt(length, 0xffffffffffffffff) {
+					data := 0
+					leave
+				}
+
+				let end := add(add(msg, 0x20), length)
+				if gt(end, add(data, returndatasize())) {
+					data := 0
+					leave
+				}
+
+				mstore(<freeMemoryPointer>, and(add(end, 0x1f), not(0x1f)))
```

Comment:
This is also wrong :(

Reply:
Or is it? The thing is that I would round up the length and not the absolute memory position, but I can also see arguments for the latter.